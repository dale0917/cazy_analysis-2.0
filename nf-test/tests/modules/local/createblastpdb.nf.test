nextflow_process {

    name "Test Process CREATEBLASTPDB"
    script "modules/local/createblastpdb.nf"
    process "CREATEBLASTPDB"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                outdir = "/workspace/nf-test/nf-core-nftest/tests/results"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = file("/workspace/nf-test/nf-test/assets/secuencias_11.txt")
                """
            }
        }
        then {
            assertAll (
                // Verificar que el proceso se ejecut√≥ sin errores
                { assert process.success },
                { assert process.out.blast_db[0][1] ==~ ".*/output_db/.*_blastp_db.phr" },
                { assert process.out.blast_db[0][2] ==~ ".*/output_db/.*_blastp_db.pin" },
                { assert process.out.blast_db[0][5] ==~ ".*/output_db/.*_blastp_db.psq" },
                { assert snapshot(process.out).match()}
            )
        }

    }

}
