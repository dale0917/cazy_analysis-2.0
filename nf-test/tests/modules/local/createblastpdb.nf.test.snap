{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        "secuencias_11_blastp_db.pdb:md5,15fe35850a2fbb327f790a663922623a",
                        "secuencias_11_blastp_db.phr:md5,dfa800a6e286c265437589277139e969",
                        "secuencias_11_blastp_db.pin:md5,93a98b9c845643732312929578c1b2d3",
                        "secuencias_11_blastp_db.pjs:md5,21874506f2f84b7f26cf4a9ec3ccccde",
                        "secuencias_11_blastp_db.pot:md5,5ab34ae1551da698d8f32f7f5659e2d2",
                        "secuencias_11_blastp_db.psq:md5,755fa4dbe16d99ee7e3101ada8e33adb",
                        "secuencias_11_blastp_db.ptf:md5,de1250813f0c7affc6d12dac9d0fb6bb",
                        "secuencias_11_blastp_db.pto:md5,1c02e5bb2051e53dbdf4c48067a5d1cd"
                    ]
                ],
                "blast_db": [
                    [
                        "secuencias_11_blastp_db.pdb:md5,15fe35850a2fbb327f790a663922623a",
                        "secuencias_11_blastp_db.phr:md5,dfa800a6e286c265437589277139e969",
                        "secuencias_11_blastp_db.pin:md5,93a98b9c845643732312929578c1b2d3",
                        "secuencias_11_blastp_db.pjs:md5,21874506f2f84b7f26cf4a9ec3ccccde",
                        "secuencias_11_blastp_db.pot:md5,5ab34ae1551da698d8f32f7f5659e2d2",
                        "secuencias_11_blastp_db.psq:md5,755fa4dbe16d99ee7e3101ada8e33adb",
                        "secuencias_11_blastp_db.ptf:md5,de1250813f0c7affc6d12dac9d0fb6bb",
                        "secuencias_11_blastp_db.pto:md5,1c02e5bb2051e53dbdf4c48067a5d1cd"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-11T18:38:29.43187"
    }
}